services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationDataMapper.kt

```diff
@@ -74,4 +81,16 @@ object ApplicationDataMapper {
             }
         }
     }
+
+    private fun path(vararg field: String) = field.joinToString(separator = ".", prefix = BASE)
+
+    private fun ZonedDateTime?.orThrow(path: String) = this ?: throw AlluDataException(path, NULL)
+
+    private fun workDescription(cableReport: CableReportApplicationData): String =
+        with(cableReport) {
+            val excavation = rockExcavation ?: throw AlluDataException(path(EXCAVATION), NULL)
+            return workDescription + excavation.description()
+        }
+
+    private fun Boolean.description(): String = if (this) "\nLouhitaan" else "\nEi louhita"
```

Comment:
I think this makes more sense as a regular function. Assigning a `description` to `Boolean` doesn't offer any hints on what it returns.

```suggestion
    private fun excavationText(excavation: Boolean) =
        if (excavation) "\nLouhitaan" else "\nEi louhita"
```

Reply:
Fixed as suggested.